R/transform-files.R

```diff
@@ -72,17 +72,37 @@ transform_file <- function(path,
 #' @inheritParams parse_transform_serialize_r
 #' @keywords internal
 #' @importFrom purrr when
-make_transformer <- function(transformers, include_roxygen_examples, warn_empty = TRUE) {
+make_transformer <- function(transformers,
+                             include_roxygen_examples,
+                             warn_empty = TRUE) {
   force(transformers)
+  cache_dir <- c("styler", cache_get_name())
+  assert_R.cache_installation(action = "warn")
   function(text) {
-    transformed_code <- text %>%
-      parse_transform_serialize_r(transformers, warn_empty = warn_empty) %>%
-      when(
-        include_roxygen_examples ~
-        parse_transform_serialize_roxygen(., transformers),
-        ~.
-      )
-    transformed_code
+    is_cached <- rlang::is_installed("R.cache") && !is.null(
+      R.cache::findCache(key = hash_standardize(text), dir = cache_dir)
+    )
+    should_use_cache <- cache_is_activated()
+    use_cache <- is_cached && should_use_cache
```

Comment:
```suggestion
    use_cache <- should_use_cache && is_cached(text)
```

with a suitable `is_cached()` function avoids calling `hash_standardize()` if not needed.